docs/ServerMaintenanceEvent.md

```diff
@@ -0,0 +1,67 @@
+
+If you are a Redis vendor and want to integrate support for ServerMaintenanceEvents into StackExchange.Redis, we recommend opening an issue so we can discuss the details.
+
+## Types of events
+
+Azure Cache for Redis currently sends the following notifications: 
+* `NodeMaintenanceScheduled`: Indicates that a maintenance event is scheduled. Can be 10-15 minutes in advance. 
+* `NodeMaintenanceStarting`: This event gets fired ~20s before maintenance begins
+* `NodeMaintenanceStart`: This event gets fired when maintenance is imminent (<5s)
+* `NodeMaintenanceFailoverComplete`: Indicates that a replica has been promoted to primary
+* `NodeMaintenanceEnded`: Indicates that the node maintenance operation is over
+
+## Sample code 
+
+The library will automatically subscribe to the pub/sub channel to receive notifications from the server, if one exists. For Azure Redis caches, this is the 'AzureRedisEvents' channel. To plug in your maintenance handling logic, you can pass in an event handler via the `ServerMaintenanceEvent` event on your `ConnectionMultiplexer`. For example:
+
+```
+multiplexer.ServerMaintenanceEvent += (object sender, ServerMaintenanceEvent e) =>
+{
+    if (e is AzureMaintenanceEvent azureEvent && azureEvent.NotificationType == AzureNotificationType.NodeMaintenanceStart)
+    {
+        // Take whatever action is appropriate for your application to handle the maintenance operation gracefully. 
+        // This might mean writing a log entry, redirecting traffic away from the impacted Redis server, or
+        // something entirely different.
+    }
+};
+```
+You can see the schema for the `AzureMaintenanceEvent` class [here](https://github.com/StackExchange/StackExchange.Redis/blob/main/src/StackExchange.Redis/Maintenance/AzureMaintenanceEvent.cs). Note that the library automatically sets the `ReceivedTimeUtc` timestamp when the event is received, so if you see in your logs that `ReceivedTimeUtc` is after `StartTimeUtc`, this may indicate that your connections are under high load.
+
+## Walking through a sample maintenance event
+
+1. App is connected to Redis and everything is working fine. 
+2. Current Time: [16:21:39] -> `NodeMaintenanceScheduled` event is raised, with a `StartTimeUtc` of 16:35:57 (about 14 minutes from current time).
+    * Note: the start time for this event is an approximation, because we will start getting ready for the update proactively and the node may become unavailable up to 3 minutes sooner. We recommend listening for NodeMaintenanceStarting and NodeMaintenanceStart for the highest level of accuracy (these are only likely to differ by a few seconds at most).
```

Comment:
```suggestion
    * Note: the start time for this event is an approximation, because we will start getting ready for the update proactively and the node may become unavailable up to 3 minutes sooner. We recommend listening for `NodeMaintenanceStarting` and `NodeMaintenanceStart` for the highest level of accuracy (these are only likely to differ by a few seconds at most).
```